### ISSUE TITLE
How to download "embedded" youtube clip?

### ISSUE THREAD

**Issue by Rava77** [yt-dlp/yt-dlp#8476]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.10.13** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I looked through the README but the link mentioned there to https://github.com/yt-dlp/yt-dlp#modifying-metadata seems not to work.

My question ( running yt-dlp version : 2023.10.13 )

When I have a downloaded video, and in it it is mentioned (pointing into some corner) "click here to get info on (whatever)" - **can I extract this embedded metadata so that I can also download the mentioned youtube clip**? I very much prefer not to have to open any videos in youtube itself and then move to the same time stamp in that video to be able to "click here to get info on (whatever)"

Can it be done via --parse-metadata ? If so, how?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by gamer191** [comment]
> I looked through the README but the link mentioned there to https://github.com/yt-dlp/yt-dlp#modifying-metadata seems not to work.

It links to a specific part of the readme. If the link won't work, just ctrl+f search for `The metadata obtained by the`.

> When I have a downloaded video, and in it it is mentioned (pointing into some corner) "click here to get info on (whatever)" - **can I extract this embedded metadata so that I can also download the mentioned youtube clip**? I very much prefer not to have to open any videos in youtube itself and then move to the same time stamp in that video to be able to "click here to get info on (whatever)"

I'm confused. Can you please send an example url?

**Comment by dirkf** [comment]
GitHub hates you, so links in user-created content only work when JS is enabled. You have to prefix the anchor tag with `user-content-` to get an actual valid HTML link, like https://github.com/yt-dlp/yt-dlp#user-content-modifying-metadata.

If the requirement is to chain to further videos mentioned in metadata, then this (not tested by me, but looks like a neat feature) from the link above seems to be the way forward:
>* You can download an additional URL based on the metadata of the currently downloaded video. To do this, set the field `additional_urls` to the URL that you want to download. E.g. `--parse-metadata "description:(?P<additional_urls>https?://www\.vimeo\.com/\d+)"` will download the first vimeo video found in the description

**Comment by Rava77** [comment]
@gamer191
Example: the video is o3BkBlWmdtE on yt.

He mentioned a video that must be embedded when you watch it on yt at time stamp 02:21 when he point to the top right corner.

**Comment by bashonly** [comment]
The information about these linked videos is not extracted by yt-dlp. So it is not currently possible. On the webpage, if you scroll down past the description and "show transcript" button, you will see all of the linked videos. You could copy the URLs from there instead of watching & waiting for the popup-video items.

Youtube calls these overlaid items "infocards", and there is a `videoDescriptionInfocardsSectionRenderer` field in the webpage JSON.
The path to the infocard video URLs is:
```jq
.videoDescriptionInfocardsSectionRenderer.infocards[].compactInfocardRenderer.content.structuredDescriptionVideoLockupRenderer.navigationEndpoint.commandMetadata.webCommandMetadata.url
```

But yt-dlp does not (yet) have a suitable info dict field to extract this information to. @coletdjnz @Grub4K is this something that should be considered in scope?

**Comment by Rava77** [comment]
_@[bashonly](https://github.com/bashonly) added site-enhancement(https://github.com/yt-dlp/yt-dlp/labels/site-enhancement)  triage(https://github.com/yt-dlp/yt-dlp/labels/triage)_
@all How likely is it that this feature will be added anytime soon? 

I would very much avoid having to open any videos in youtube at all. Currently the video would load and play, but the rest of the page won't (so no description for me, and also no '"show transcript" button'), I presume due to how youtube fights back, code-wise, against uBlockOrigin).
Alternatively tried opening the same video via https://yewtu.be/ but I do not see any  '"show transcript" button' below the description.

**Comment by coletdjnz** [comment]
Are these considered "recommended" videos? iirc we've considered that out of scope for the project - yt-dlp isn't a YouTube player client.

If they are considered "part" of the video or sources so to say then it may be similar to https://github.com/yt-dlp/yt-dlp/issues/4217#issuecomment-1345691508. Where we can have an attributions-like metadata. Though this will require some design work.

**Comment by Rava77** [comment]
No, they are not recommended videos. These are inserted by the video maker, not by yt.

**Comment by bashonly** [comment]
They are unlisted videos created specifically as supplements to the main video. IMO this is in scope, we just need to figure out the best way to implement support

**Comment by Rava77** [comment]
"_In the OP video's case, they are unlisted videos created specifically as supplements to the main video._"

Meaning in a case like m example, I will have to play the video and copy all the embedded links, because I would not get the URLs / the yt-codes of the videos otherwise for now.

**Comment by observeroftime01** [comment]
How would such a feature behave? Let's say I specify that I want to download those infocard videos as well, and the video refereed to by the infocard also contains yet another infocard referring to yet another video, and so on. A user should be able to specify a maximum "depth" to fetch those videos to prevent unlimited recursion.

For example, what if Video A points at Video B, which points at video C, which points at video D, which points back at video A? Granted, I don't know any videos that have such circular referencing, but such a scenario should probably still be accounted for. If needed I could probably craft such a circular series of videos to test against.

**Comment by dirkf** [comment]
Isn't this a problem that's already solved for playlists?

If modelled as a playlist, an infocard would be a special type with additional metadata. Ideally a solution for YT should take account of similar scenarios for other sites and of applicable standards, not that I'm aware of any.

**Comment by pukkandan** [comment]
We can implement the extraction as per https://github.com/yt-dlp/yt-dlp/issues/4217#issuecomment-1345691508 and then it will be possible to use `additional_urls` to download them. Adding an easier way to download it could be considered separately based on demand.


Semi-related, should we also treat premier trailers as attributions?

Related: https://github.com/yt-dlp/yt-dlp/issues/8390 https://github.com/yt-dlp/yt-dlp/issues/5291 https://github.com/yt-dlp/yt-dlp/pull/5899

**Comment by Rava77** [comment]
@pukkandan "_We can implement the extraction as per https://github.com/yt-dlp/yt-dlp/issues/4217#issuecomment-1345691508 and then it will be possible to use `additional_urls` to download them. Adding an easier way to download it could be considered separately based on demand._"

I do not recall ever having used `additional_urls`, but when it will work, and when I do not have to watch the video on yt to note down the embedded "infocard" videos, then I am a happy camper.
